cupy/lib/polynomial.pyx

```diff
@@ -0,0 +1,338 @@
+            self._coeffs = cupy.concatenate((zeroz, self.coeffs))
+            ind = 0
+        self._coeffs[ind] = val
+        return
+
+    def __iter__(self):
+        return iter(self.coeffs)
+
+    def integ(self, m=1, k=0):
+        raise NotImplementedError
+
+    def deriv(self, m=1):
+        raise NotImplementedError
+
+    # -------------------------------------------------------------------------
+    # Cupy specific attributes and methods
+    # -------------------------------------------------------------------------
+    def get(self, stream=None, out=None):
+        """Returns a copy of poly1d object on host memory.
+
+        Args:
+            stream (cupy.cuda.Stream): CUDA stream object. If it is given, the
+                copy runs asynchronously. Otherwise, the copy is synchronous.
+                The default uses CUDA stream object of the current context.
+            out (numpy.poly1d): Output object. In order to enable asynchronous
+                copy, the underlying memory should be a pinned memory.
+
+        Returns:
+            numpy.poly1d: Copy of poly1d object on host memory.
+
+        """
+        if out is not None:
+            if not isinstance(out, numpy.poly1d):
+                raise TypeError('Only numpy.poly1d can be obtained from'
+                                'cupy.poly1d')
+            if self.coeffs.dtype != out.coeffs.dtype:
+                raise TypeError(
+                    '{} poly1d cannot be obtained from {} poly1d'.format(
+                        out.coeffs.dtype, self.coeffs.dtype))
+            if self.coeffs.shape != out.coeffs.shape:
+                raise ValueError(
+                    'Shape mismatch. Expected shape: {}, '
+                    'actual shape: {}'.format(self.coeffs.shape,
+                                              out.coeffs.shape))
+            coeffs_cpu = out.coeffs
+        else:
+            if self.coeffs.size == 0:
+                c = numpy.array(self.coeffs.shape, self.coeffs.dtype)
+                return numpy.poly1d(c, variable=self.variable)
+            coeffs_gpu = self.coeffs
+            coeffs_cpu = numpy.empty(self.coeffs.shape, self.coeffs.dtype)
+        syncdetect._declare_synchronize()
+        ptr = ctypes.c_void_p(coeffs_cpu.__array_interface__['data'][0])
+        with self.coeffs.device:
+            if stream is not None:
+                coeffs_gpu.data.copy_to_host_async(ptr,
+                                                   coeffs_gpu.nbytes, stream)
+            else:
+                stream_ptr = stream_module.get_current_stream_ptr()
+                if stream_ptr == 0:
+                    coeffs_gpu.data.copy_to_host(ptr, coeffs_gpu.nbytes)
+                else:
+                    coeffs_gpu.data.copy_to_host_async(ptr, coeffs_gpu.nbytes)
+        return numpy.poly1d(coeffs_cpu, variable=self.variable)
+
+
+warnings.simplefilter('always', numpy.RankWarning)
```

Comment:
This line is not needed because `numpy.RankWarning` will be enabled to `always` when importing `numpy`.